HARK/distribution.py

```diff
@@ -930,29 +929,288 @@ def draw(self, N, X=None, exact_match=False):
+    stored in an underlying `xarray.DataArray` array.
+
+    Parameters
+    ----------
+    pmf : np.array
+        An array of floats representing a probability mass function.
+    data : np.array
+        Discrete point values for each probability mass.
+        For multivariate distributions, the last dimension of X must index
+        "nature" or the random realization. For instance, if X.shape == (2,6,4),
+        the random variable has 4 possible realizations and each of them has shape (2,6).
+    seed : int
+        Seed for random number generator.
+    coords : dict
+        Coordinate values/names for each dimension of the underlying array.
+    dims : tuple or list
+        Dimension names for each dimension of the underlying array.
+    name : str
+        Name of the distribution.
+    attrs: dict
+        Attributes for the distribution.
+    """
+
+    def __init__(
+        self,
+        pmf,
+        data,
+        seed=0,
+        coords=None,
+        dims=None,
+        name="DiscreteDistributionXRA",
+        attrs=None,
+    ):
+
+        if data.ndim < 2:
+            data = data[np.newaxis, ...]
+
+        if attrs is None:
+            attrs = {}
+
+        attrs["pmf"] = np.asarray(pmf)
+        attrs["seed"] = seed
+        attrs["RNG"] = np.random.RandomState(seed)
+
+        self._xarray = DataArray(
+            data=data,
+            coords=coords,
+            dims=dims,
+            name=name,
+            attrs=attrs,
+        )
+
+    @property
+    def xarray(self):
+        """
+        Returns the underlying xarray.DataArray object.
+        """
+        return self._xarray
+
+    @property
+    def X(self):
+        """
+        Returns the distribution's data as a numpy.ndarray.
+        """
+        return self._xarray.values
+
+    @property
+    def pmf(self):
+        """
+        Returns the distribution's probability mass function.
+        """
+        return self._xarray.pmf
+
+    @property
+    def RNG(self):
+        """
+        Returns the distribution's random number generator.
+        """
+        return self._xarray.RNG
+
+    @property
+    def data(self):
+        """
+        The distribution's data as an array. The underlying
+        array type (e.g. dask, sparse, pint) is preserved.
+        """
+        return self._xarray.data
+
+    @property
+    def coords(self):
+        """
+        The distribution's coordinates.
+        """
+        return self._xarray.coords
+
+    @property
+    def dims(self):
+        """
+        The distribution's dimensions.
+        """
+        return self._xarray.dims
+
+    @property
+    def name(self):
+        """
+        The distribution's name.
+        """
+        return self._xarray.name
+
+    @property
+    def attrs(self):
+        """
+        The distribution's attributes.
+        """
+        return self._xarray.attrs
+
+    def expected_value(self, func=None, *args, labels=False):
```

Comment:
The relevant docs on argument ordering are here:
https://docs.python.org/3/tutorial/controlflow.html?highlight=arguments%20style%20positional%20default#special-parameters